bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/discovery/HueBridgeMDNSDiscoveryParticipant.java

```diff
@@ -0,0 +1,136 @@
+import org.openhab.core.config.discovery.mdns.MDNSDiscoveryParticipant;
+import org.openhab.core.config.discovery.mdns.internal.MDNSDiscoveryService;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingTypeUID;
+import org.openhab.core.thing.ThingUID;
+import org.osgi.service.component.ComponentContext;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Modified;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link HueBridgeMDNSDiscoveryParticipant} is responsible for discovering new and removed hue bridges. It uses the
+ * central {@link MDNSDiscoveryService}.
+ *
+ * @author Kai Kreuzer - Initial contribution
+ * @author Thomas Höfer - Added representation
+ * @author Christoph Weitkamp - Change discovery protocol to mDNS
+ */
+@Component(configurationPid = "discovery.hue")
+@NonNullByDefault
+public class HueBridgeMDNSDiscoveryParticipant implements MDNSDiscoveryParticipant {
+
+    private static final String SERVICE_TYPE = "_hue._tcp.local.";
+    private static final String MDNS_PROPERTY_BRIDGE_ID = "bridgeid";
+    private static final String MDNS_PROPERTY_MODEL_ID = "modelid";
+
+    private static final String REMOVAL_GRACE_PERIOD = "removalGracePeriod";
+
+    private final Logger logger = LoggerFactory.getLogger(HueBridgeMDNSDiscoveryParticipant.class);
+
+    // Hue bridges have maxAge 100 seconds, so set the default grace period to half of that
+    private long removalGracePeriodSeconds = 50;
+
+    private boolean isAutoDiscoveryEnabled = true;
+
+    @Activate
+    protected void activate(ComponentContext componentContext) {
+        activateOrModifyService(componentContext);
+    }
+
+    @Modified
+    protected void modified(ComponentContext componentContext) {
+        activateOrModifyService(componentContext);
+    }
+
+    private void activateOrModifyService(ComponentContext componentContext) {
+        Dictionary<String, @Nullable Object> properties = componentContext.getProperties();
+        String autoDiscoveryPropertyValue = (String) properties
+                .get(DiscoveryService.CONFIG_PROPERTY_BACKGROUND_DISCOVERY);
+        if (autoDiscoveryPropertyValue != null && !autoDiscoveryPropertyValue.isBlank()) {
+            isAutoDiscoveryEnabled = Boolean.valueOf(autoDiscoveryPropertyValue);
+        }
+        String removalGracePeriodPropertyValue = (String) properties.get(REMOVAL_GRACE_PERIOD);
+        if (removalGracePeriodPropertyValue != null && !removalGracePeriodPropertyValue.isBlank()) {
+            try {
+                removalGracePeriodSeconds = Long.parseLong(removalGracePeriodPropertyValue);
+            } catch (NumberFormatException e) {
+                logger.warn("Configuration property '{}' has invalid value: {}", REMOVAL_GRACE_PERIOD,
+                        removalGracePeriodPropertyValue);
+            }
+        }
+    }
```

Comment:
I am wondering if the mDNS discovery for Hue Bridges suffer from this behavior at all. In https://github.com/openhab/openhab-addons/pull/9985 we introduced the grace period because of a wrong (or too low) `maxAge` value for UPnP discovery. What is the matching criteria for mDNS? Maybe we should spent some more time to test my replacement and decide afterwards if we still need this configuration for the Hue binding? Wdyt?

Reply:
Exactly.

Reply:
@cweitkamp - good point. It could also be kept, but with a default value of 0.

Reply:
Hmm, now just brainstorming: Would it be possible for MDNSDiscoveryParticipant (in core) to simply read that configuration taking the binding id into consideration? I.e. get value of `discovery.<binding id>:removalGracePeriod` and apply that if it exists? Then bindings would only have to implement anything if they need a default value != 0. In all cases the .cfg value would override binding default (which would be 0 if binding has no default).

Reply:
Hmm.. I am not sure if the MDNSDiscoveryParticipant in core has a way (currently) to know the binding Id. Especially if we want to implement @kaikreuzer's suggestion of pre-discovering Things on new systems before any bindings have been installed. And if it would be possible then we would obviously have to revert both mDNS and UPnP disco..

Reply:
@andrewfg - can this comment be closed? I'm not sure if anything is left to do in this PR, putting aside some general ideas and future wishes. :-)